### PR TITLE
Documented env vars and generally updated README

### PR DESCRIPTION
**Pull request by olsonadr**
Here's your PR with the required secrets, @smcgregor!

Specifically updated wording and fixed broken links in the README, moved the image into the docs dir, moved the sentiment.py example lambda into the docs/example_lambdas dir, removed old and innacurate amazon licensing accidentally left in from archived scratch work based on sample code, and moved some of the gratuitous detail from the README.md file into a supplemental docs/gratuitously_detailed_explanations.md file.

### CODE REVIEW

File: inference/similar.py [hunk 1]
```diff
@@ -1,8 +1,3 @@
-"""
-Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-SPDX-License-Identifier: MIT-0
```
Comment (smcgregor) [inline inference/similar.py:3]: What made you strip the license statement? Is this not derived from Amazon code?

File: docs/example_lambdas/sentiment.py [hunk 2]
```diff
@@ -1,8 +1,3 @@
-"""
-Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
-SPDX-License-Identifier: MIT-0
```
Comment (smcgregor) [inline docs/example_lambdas/sentiment.py:3]: Why stripped?

**Comment by smcgregor** [comment]
Thanks @olsonadr, I skimmed it and had a question on the license removal on two files. Full review will require a bit longer since the review should be done by stepping through the integration. ETA ~1 week.

I enjoyed *gratuitously*_detailed_explanations.md as a filename.

File: inference/similar.py (hunk 1 above)
Comment (olsonadr) [reply inference/similar.py:3]: The derivation was: we started from the sample code, completely replaced 99% of that code with our own, then recombined chunks of our code into a new source file later on, and the license got copied in with our imports. We thought we had talked through it with you but maybe it fell off the agenda. We're open to leaving them in these files on your judgement

File: docs/example_lambdas/sentiment.py (hunk 2 above)
Comment (olsonadr) [reply docs/example_lambdas/sentiment.py:3]: It got stripped along with the similar.py statement, but I think this one should stay in. It is a direct path of derivation to get here

**Comment by olsonadr** [comment]
> 

I'm glad you enjoyed it! Let us know what you think on the license diffs and we can fix as needed

**Pull request merged**